doio.c

```diff
@@ -2259,7 +2260,7 @@ Perl_do_print(pTHX_ SV *sv, PerlIO *fp)
          * but only until the system hard limit/the filesystem limit,
          * at which we would get EPERM.  Note that when using buffered
          * io the write failure can be delayed until the flush/close. --jhi */
-        if (len && (PerlIO_write(fp,tmps,len) == 0))
+        if (len && (PerlIO_write(fp,(char *) tmps,len) == 0))
```

Comment:
PerlIO_write() takes a `const void *` for the buffer, you don't need the cast.

Reply:
This is still the case, no cast is required, PerlIO_write().

Reply:
sorry; I somehow missed seeing your first comment